src/main/scala/higherkindness/skeuomorph/mu/comparison/Comparison.scala

```diff
@@ -182,8 +182,11 @@ object Comparison extends ComparisonInstances {
           val nestedCoproducts = zipLists(path, nc, nc2, _ => Name(n))
           CompareList(fields ++ nestedProducts ++ nestedCoproducts)
 
-        // Numeric widdening
-        case (TInt(), TLong() | TFloat() | TDouble()) | (TLong(), TFloat() | TDouble()) | (TFloat(), TDouble()) =>
+        // Numeric widening
+        // TODO: this is not valid for Protobuf or Avro (e.g. changing an int to a float would be a breaking change)
+        case (x: TInt[_], y: TInt[_]) if x.size == _32 && y.size == _64 =>
```

Comment:
Would this work?

```suggestion
        case (TInt(_32), TInt(_64)) =>
```

Probably we need to implement a `TInt.unapply` for it.